### PR TITLE
FUSETOOLS-1789 - Don't remove all data formats

### PR DESCRIPTION
**Pull request by apupier**
I didn't find any use cases which was handled by the code i removed. @lhein do you have some in mind?

### CODE REVIEW

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java [hunk 1]
```diff
@@ -626,22 +626,6 @@ public void setParameter(String name, Object value) {
 	}
 
 	/**
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java]: I'd keep the code for now.

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java [hunk 2]
```diff
@@ -669,8 +653,6 @@ protected void setParameter(String name, Object value, boolean overrideChangeChe
 		
 		if (value instanceof AbstractCamelModelElement && getParameter("ref") != null) {
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java]: Lets remove the whole if .. if else

**Comment by lhein** [comment]
Please add some validation so the user gets aware he is needed to choose between ref and a direct dataformat definition if we don't correct that any more automatically.

**Comment by apupier** [comment]
I added the validation. Can you review it @lhein ?

File: core/plugins/org.fusesource.ide.camel.validation/src/org/fusesource/ide/camel/validation/model/TextParameterValidator.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	@Override
+	public IStatus validate(Object value) {
+		if (PropertiesUtils.isRequired(parameter) || parameter.getName().equalsIgnoreCase("id")) {
+			final Object uriValue = camelModelElement.getParameter("uri");
+			final Object refValue = camelModelElement.getParameter("ref");
+			if (parameter.getName().equalsIgnoreCase("uri")) {
+				// only enforce URI if there is no REF set
+				if (uriValue == null || ((String) uriValue).trim().length() < 1) {
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.validation/src/org/fusesource/ide/camel/validation/model/TextParameterValidator.java]: Strings.isBlank()?

File: core/plugins/org.fusesource.ide.camel.validation/src/org/fusesource/ide/camel/validation/model/TextParameterValidator.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+					// no URI set -> check for REF
+					if (refValue == null || ((String) refValue).trim().length() < 1) {
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.validation/src/org/fusesource/ide/camel/validation/model/TextParameterValidator.java]: Strings.isBlank()?

**Comment by lhein** [comment]
Commented 2 lines. Rest looks fine to me.

**Pull request merged**